docs/rfc/rfc-015-abci++-tx-mutation.md

```diff
@@ -0,0 +1,261 @@
+# RFC 015: ABCI++ TX Mutation
+
+## Changelog
+
+- 23-Feb-2022: Initial draft (@williambanfield).
+- 28-Feb-2022: Revised draft (@williambanfield).
+
+## Abstract
+
+The specification of ABCI++ details a mechanism for proposers to replace transactions
+in the proposed block. This scheme requires the proposer to construct new transactions
+and mark these new transactions as replacing other removed transactions. The specification
+is ambiguous as to how the replacement may be communicated to peer nodes.
+This RFC discusses issues with this mechanism and possible solutions.
+
+## Background
+
+### What is the proposed change?
+
+The ABCI++ specification proposes mechanisms for adding, removing, and replacing
```

Comment:
We have #7663 in the board to deal with this. A few weeks ago, it became clear that we had to change the current ABCI++ spec (hence #7763).
I'd suggest to change the wording of this paragraph to refer to "a previous version of the ABCI++ spec"; as #8002 (tackled by @angbrav) will remove `new_hashes` from the spec.

Reply:
Updated.